packages/patternfly-3/patternfly-react-wooden-tree/src/components/TreeView/TreeViewExample.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import TreeView from './index';
+
+const data = [
+  {
+    text: 'Parent 1',
+    icon: 'fa fa-folder',
+    nodes: [
+      {
+        text: 'Child 1',
+        nodes: [{ text: 'Grandchild 1' }, { text: 'Grandchild 2' }]
+      },
+      {
+        text: 'Child 2',
+        nodes: [{ text: 'Grandchild 1' }, { text: 'Grandchild 2' }]
+      }
+    ]
+  },
+  {
+    text: 'Parent 2',
+    icon: 'fa fa-folder',
+    nodes: [{ text: 'Child 1' }]
+  },
+  {
+    text: 'Parent 3',
+    icon: 'fa fa-folder'
+  },
+  {
+    text: 'Parent 4',
+    icon: 'fa fa-folder'
+  },
+  {
+    text: 'Parent 5',
+    icon: 'fa fa-folder'
+  }
+];
+
+const actionMapper = {
+  'state.expanded': TreeView.nodeExpanded,
+  'state.checked': TreeView.nodeChecked,
+  'state.disabled': TreeView.nodeDisabled,
+  'state.selected': TreeView.nodeSelected,
+  nodes: TreeView.nodeChildren,
+  loading: TreeView.nodeLoading
+};
+
+class TreeViewExample extends React.Component {
+  /**
```

Comment:
```suggestion
  /**
    data = TreeView.initTree(data);

  state = {
    tree: this.data
  };

  /**
   * The callback function for changing data in the tree.
   *
   * @param {string} nodeId The nodeId of the node.
   * @param {string} type The field name which changed.
   * @param {boolean} value The new value to assign.
   */

  onDataChange = (nodeId, type, value) => {
    console.log('hi');
    let node = TreeView.nodeSelector(this.data, nodeId);
    if (node == null) {
      return;
    }

    if (actionMapper.hasOwnProperty(type)) {
      node = actionMapper[type](node, value);
      this.data = TreeView.nodeUpdater(this.data, node);
    }
    this.setState({ tree: this.data });
  };
```

Reply:
Nice work! Can we remove the constructor and the bound function definition? Since we are using class properties, we can just add the state & data values right on the TreeViewExample class.

Reply:
Actually, we need the bind method and the state in the constructor because of the react. Your suggestion is not working.

Reply:
We are using class properties [1] so you don't really need to use bind.

Is there any particular reason why to use `this.data`? You shouldn't really use attributes in react components unless you have some reasoning behind this. If I am reading it correctly you will emit re-render with same tree data if type was not found in action mapper?

[1] https://babeljs.io/docs/en/babel-plugin-proposal-class-properties

Reply:
@karelhala you are right there is no real reason to use this.data. We could just assign the data to a variable & update it that way. 
@brumik I did test the removal of the constructor locally & it worked. If you remove the bind method, you can accomplish the same thing by using an arrow function for onDataChange. Arrow Functions lexically bind their context so this actually refers to the originating context.